packages/zendesk-client/src/use-smooch.ts

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import Smooch from 'smooch';
+import { SMOOCH_CONTAINER_ID, SMOOCH_INTEGRATION_ID } from './constants';
+import { useAuthenticateZendeskMessaging } from './use-authenticate-zendesk-messaging';
+
+export const useSmooch = () => {
+	const [ init, setInit ] = useState(
+		!! document.getElementById( SMOOCH_CONTAINER_ID )?.querySelector( 'iframe' )
+	);
+	const { data: authData } = useAuthenticateZendeskMessaging( true, 'messenger' );
+	const [ addMessage, setAddMessage ] =
+		useState< ( message: { text: string; role: string } ) => void >();
+
+	useEffect( () => {
+		const messengerContainer = document.getElementById( SMOOCH_CONTAINER_ID );
+		if ( authData?.jwt && authData?.externalId && ! init ) {
+			Smooch.init( {
+				integrationId: SMOOCH_INTEGRATION_ID,
+				embedded: true,
+				externalId: authData.externalId,
+				jwt: authData.jwt,
+			} ).then( () => {
+				setInit( true );
+			} );
+			if ( messengerContainer ) {
+				Smooch.render( messengerContainer );
+			}
+		}
+	}, [ authData?.externalId, authData?.jwt, init ] );
+
+	if ( ! init ) {
+		return {
+			init: false,
+			destroy: () => {},
+			getConversation: () => undefined,
+			createConversation: () => {},
+			addMessengerListener: () => {},
+			sendMessage: () => {},
+		};
+	}
+
+	const getConversation = async ( chatId?: number ): Promise< Conversation | undefined > => {
```

Comment:
Do we need to check `init` below this line? We already exited early when it's false.